tests/kernel/mutex/mutex_api/src/test_mutex_apis.c

```diff
@@ -91,7 +108,99 @@ static void tmutex_test_lock_unlock(struct k_mutex *pmutex)
 	k_mutex_unlock(pmutex);
 }
 
+static void tThread_T1_priority_inheritance(void *p1, void *p2, void *p3)
+{
+	/* t1 will get mutex first */
+	zassert_true(k_mutex_lock((struct k_mutex *)p1, K_FOREVER) == 0,
+		      "access locked resource from spawn thread T1");
+
+	/* record its original priority */
+	int priority_origin = k_thread_priority_get((k_tid_t)p2);
+
+	TC_PRINT("T1 is going to enter mutex: origin priority=%d\n",
+		priority_origin);
+
+	/* wait for a time period to see if priority inheritance happended */
+	k_sleep(K_MSEC(500));
+
+	int priority = k_thread_priority_get((k_tid_t)p2);
+
+	TC_PRINT("access resource from thread T1: priority=%d\n", priority);
+
+	zassert_true(priority <= priority_origin,
+		      "priority inheritance not happened!");
+
+	/* For testing case 2 */
+	if (p3 != NULL) {
+		zassert_true(priority <= priority_origin,
+				"fail to priority inheritance not happened!");
+
+		/* wait for t2 timeout to get mutex*/
+		k_sleep(K_MSEC(TIMEOUT));
+
+		k_mutex_unlock((struct k_mutex *)p1);
+	}
+	/* For testing case 1 and 3 */
+	else {
+		if (priority <= priority_origin) {
+			k_mutex_unlock((struct k_mutex *)p1);
+		}
+
+		/* check if priority set back to original one */
+		priority = k_thread_priority_get((k_tid_t)p2);
+		TC_PRINT("after release thread T1: priority=%d\n", priority);
+
+		zassert_true(priority == priority_origin,
+			"priority inheritance adjust back not happened!");
+	}
+}
+
+static void tThread_T2_priority_inheritance(void *p1, void *p2, void *p3)
+{
+	TC_PRINT("thread T2 priority=%d\n",
+			k_thread_priority_get((k_tid_t)p2));
+
+	/* For testing case 2 */
+	if (p3 != NULL) {
+		zassert_false(k_mutex_lock((struct k_mutex *)p1,
+				K_MSEC(100)) == 0,
+				"fail to should not get the resource");
+
+		TC_PRINT("successful: do not get resource\n");
+	}
+	/* For testing case 1 and 3 */
+	else {
+		zassert_true(k_mutex_lock((struct k_mutex *)p1, K_FOREVER) == 0,
+		      "access locked resource from spawn thread T2");
+
+		TC_PRINT("thread T2 got the resource\n");
+
+		k_mutex_unlock((struct k_mutex *)p1);
+	}
+}
+
+static void tThread_lock_with_timeout(void *p1, void *p2, void *p3)
+{
+	zassert_true(k_mutex_lock((struct k_mutex *)p1, K_FOREVER) == 0,
+		      "access locked resource from spawn thread");
+
+	/* This thread will hold mutex for 2000 ms, then release it */
+	k_sleep(K_MSEC(2000));
```

Comment:
Two seconds is a long time.  Long-running tests in CI slow everyone down.  Can't you do this with a faster timeout?

Reply:
Yes, It's actually too long. So I adjusted it to TIMEOUT+100 (600ms) and modified relative part.